tjeaneric/Flashcards-API
# Ft authentication

#### What does this PR do?
* Implements Authentication 


#### Description of Task to be completed?
* Implements Authentication and protects some operations on unauthenticated users
* Adds `Signup` and `Login` feature

#### How should this be manually tested?
* clone the repo using SSH: `git clone git@github.com:tjeaneric/Flashcards-API.git`
                          or    HTTPS: `git clone https://github.com/tjeaneric/Flashcards-API.git`
* checkout to this branch by running `git checkout ft-authentication`
* run `npm install`
* check env.example file and add required environment variables to your .env file
* Do migrations with `npx prisma migrate dev --name "init"`
* run `npm run dev`
* got to your localhost:{port}, you will see your apollo server studio and perform operations using Mutations and Queries

#### Any background context you want to provide?
* Used `JSON web token` to generate the token for authenticated users.
* Used `bcryptjs` for hashing the password


#### Questions:
No.
